plugins/BEdita/Core/src/Model/Table/ObjectTypesTable.php

```diff
@@ -459,4 +460,26 @@ protected function findByRelation(Query $query, array $options = [])
         // Everything is said and done by now. Fingers crossed!
         return $query->where($conditionsBuilder);
     }
+
+    /**
+     * Finder to get object type starting from object id or uname.
+     *
+     * @param \Cake\ORM\Query $query Query object.
+     * @param array $options Additional options. The `id` key is required.
+     * @return \Cake\ORM\Query
+     * @throws \LogicException When missing required parameters.
+     */
+    protected function findObjectId(Query $query, array $options = [])
+    {
+        if (empty($options['id'])) {
+            throw new \LogicException(__d('bedita', 'Missing required parameter "{0}"', 'id'));
+        }
+
+        return $query->innerJoinWith('Objects', function (Query $query) use ($options) {
+            return $query->where(function (QueryExpression $exp) use ($options) {
+                return $exp->or_([$this->Objects->aliasField('id') => $options['id']])
+                    ->add([$this->Objects->aliasField('uname') => $options['id']]);
```

Comment:
I'd find slightly more readable this notation:
```php
return $exp->or_([
    $this->Objects->aliasField('id') => $options['id'],
    $this->Objects->aliasField('uname') => $options['id'],
]);
```

Reply:
More readable, sure. I didn't understand that I can do it.